### PR TITLE
Add a Dockerfile to enable building Docker containers of the SSM and document it's usage

### PR DESCRIPTION
**Pull request by gregcorbett**
Once merged, I can configure automatic builds for branches and tags so an image will get built automatically for them.

This PR (and presumably subsequent PRs) triggered a Docker build with the output shown with the rest of the PR time checks.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -121,13 +121,44 @@ add your messages.
...
+ ```
+ docker run \
+     -d --entrypoint ssmsend \
+     -v /path/to/downloaded/config/sender.cfg:/etc/apel/sender.cfg \
+     -v /path/to/read/messages:/var/spool/apel/outgoing \
+     -v /etc/grid-security:/etc/grid-security \
+     -v /path/to/persistently/log:/var/log/apel \
+     stfc/ssm
+ ```
+ * The line `-v /path/to/persistently/log:/var/log/apel \` is only required if you want to access the sender logs as a file. If `console: true` is set in your `sender.cfg`, the container will also log to stdout/stderr.
```
Comment (tofu-rocketry) [inline README.md]: Should that be "sender log" or "sender logs"?

File: README.md [hunk 2]
```diff
@@ -121,13 +121,44 @@ add your messages.
...
+ docker run \
+     -d --entrypoint ssmreceive \
+     -v /path/to/downloaded/config/sender.cfg:/etc/apel/sender.cfg \
+     -v /path/to/read/messages:/var/spool/apel/outgoing \
+     -v /path/to/dns/file:/etc/apel/dns \
+     -v /etc/grid-security:/etc/grid-security \
+     -v /path/to/persistently/log:/var/log/apel \
+     stfc/ssm
+ ```
+  * The line `-v /path/to/persistently/log:/var/log/apel \` is only required if you want to access the receiver logs as a file. If `console: true` is set in your `sender.cfg`, the container will also log to stdout/stderr.
```
Comment (tofu-rocketry) [inline README.md]: Should that be "receiver log" or "receiver logs"?

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,29 @@
+FROM centos:7
+MAINTAINER APEL Administrator <apel-admins@stfc.ac.uk>
```
Comment (tofu-rocketry) [inline Dockerfile]: "Administrators"

File: README.md (hunk 1 above)
Comment (gregcorbett) [reply README.md]: "sender log", changed

File: README.md (hunk 2 above)
Comment (gregcorbett) [reply README.md]: "receiver log", changed. I also noticed it should be `receiver.cfg` rather than `sender.cfg`.

File: Dockerfile (hunk 3 above)
Comment (gregcorbett) [reply Dockerfile]: changed

**Comment by gregcorbett** [comment]
The docker build seemed to fail according to the PR, but looking on docker cloud it had passed. I have re-triggered the build to see if that "solves" the problem.

**Comment by gregcorbett** [comment]
it did, huzzah! Docker Cloud may have gotten confused because I rather quickly amended a commit after triggering the build.

**Comment by tofu-rocketry** [comment]
Ah, yes, that can break things with CI systems sometimes.

**Comment by tofu-rocketry** [comment]
The changes are simple so I'm thinking I'll just use GitHub to squash them all together and stick on top of the dev branch. You ok with that?

**Comment by gregcorbett** [comment]
Fine by me

**Pull request merged**